pkg/mimirtool/commands/alerts_test.go

```diff
@@ -41,3 +42,43 @@ func TestReadAlertmanagerConfigTemplates_collidingNames(t *testing.T) {
 	_, err := cmd.readAlertManagerConfigTemplates()
 	require.Error(t, err)
 }
+
+func TestReadAlertmanagerValidConfigFile(t *testing.T) {
+	cmd := AlertmanagerCommand{
+		AlertmanagerConfigFile: "./testdata/alertmanager_config.yaml",
+	}
+
+	_, _, err := cmd.readAlertManagerConfig()
+	require.NoError(t, err)
+}
+
+func TestLoadAlertmanagerConfigAndTemplateFile(t *testing.T) {
+	// creates a temp directory to persist config and templates to disk
+	dir := t.TempDir()
+	// reads the existing config and template to memory
+	cmd := AlertmanagerCommand{
+		AlertmanagerConfigFile: "./testdata/alertmanager_config.yaml",
+		TemplateFiles: []string{
+			"./testdata/alertmanager_template.tmpl",
+		},
+		OutputDir: dir,
+	}
+	cfg, templates, err := cmd.readAlertManagerConfig()
+	require.NoError(t, err)
+	// write the data to a temp directory and reads it again
+	err = cmd.outputAlertManagerConfigTemplates(cfg, templates)
+	require.NoError(t, err)
+	// loads the new files that has been written out from memory, load the new files and compare
+	cmdLoad := AlertmanagerCommand{
+		AlertmanagerConfigFile: filepath.Join(dir, "config.yaml"),
+		TemplateFiles: []string{
+			filepath.Join(dir, "alertmanager_template.tmpl"),
+		},
+	}
+	cfgResult, templatesResult, err := cmdLoad.readAlertManagerConfig()
+	require.NoError(t, err)
+
+	assert.Equal(t, cfg, cfgResult)
+	assert.Equal(t, len(templates), len(templatesResult))
+	assert.Equal(t, templates, templatesResult)
```

Comment:
Templates is actually a map rather than a slice, of `map[<fileName>][fileContent]`. However, on second thought this comparison is probably a bit weak for the future so I have updated it and compare element by element.

Reply:
oh, i thought that `ElementsMatch` works for maps too. Your approach looks good too

Reply:
As a first time contributor, thanks a lot for guiding me through this PR!